File: Jogo ruby/joguin.py
import random
from sre_constants import REPEAT
from timeit import repeat

maozinha1 = []
maozinha2 = []
acao1 = 0
acao2 = 1
mesa = []
baralho = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
run = True
ult1 = 0
ult2 = 0

#########################################

def jogarcarta(jogador):
    escolha = int
    if jogador == 1:
        print(f'\033[1;91m{maozinha1}')
        while (escolha not in maozinha1):
            escolha = int(input("Escolha uma carta: "))
        mesa.append(escolha)
        maozinha1.remove(escolha)
        print(mesa)
    elif jogador == 2:
        print(f'\033[1;91m{maozinha2}')
        while (escolha not in maozinha2):
            escolha = int(input("Escolha uma carta: "))
        mesa.append(escolha)
        maozinha2.remove(escolha)
        print(mesa)
        
def turno1():
    global acao1, ult1
    print(f'\033[1;91m{maozinha1}')
    while acao1 != 1:
        acao1 = int(input('Digite 1 para comprar carta\ndigite 2 para passar\n'))
    if acao1 == 1 and ult1 != 1:
        jogarcarta(1)
    else: 
        acao1=0
    ult1 = acao1

def turno2():
    global acao2, ult2
    print(f'\033[0:36:40m{maozinha2}')
    while acao2 != 1 or 2:
        acao2 = int(input('Digite 1 para comprar carta\ndigite 2 para passar\n'))
    if acao2 == 1 and ult2 != 1:
        jogarcarta(2)
    else: 
        acao2=0
    ult2= acao2

def gerar_maos(num):
    for x in range(num):
        carta_A = random.choice(baralho)
        maozinha1.append(carta_A)
        baralho.remove(carta_A)
        carta_A = random.choice(baralho)
        maozinha2.append(carta_A)
        baralho.remove(carta_A)
    
while run:
    gerar_maos(5)
    while True:
        if acao1 == 0:
            turno1()
        elif acao1 == 0:
            turno2()

File: Jogo ruby/joguin.rb
#poker de cria 

$maozinha1 = []
$maozinha2 = []
$naipes = ['♣️', '♥', '♠', '♦️']
$valours = ['A', 2, 3, 4, 5, 6, 7, 8, 9, 10, 'J', 'Q', 'K']
$baralho = $valours.product($naipes)
$baralho = $baralho
$cartinhas = []
puts "deseja embaralhar quantas vezes? "
num = gets.to_i

cont = 0

while cont <= num
    $baralho.shuffle!
    cont += 1
end

def mao1()
    $maozinha1 = $baralho.sample(3)
    $baralho = $baralho - $maozinha1
    for $a_1 in $maozinha1
    #    p $a_1
    end
end

def mao2()
    $maozinha2 = $baralho.sample(3)
    $baralho = $baralho - $maozinha2
 #   p $maozinha2
end

cartinhas = []

def dealer()
    $mesa = $baralho.sample(5)
    $baralho = $baralho - $mesa
end

def pop()
    $x = $mesa.first
    $mesa.delete_at(0)
#    print $x
end

def tonhos()
    card = $maozinha1 + $mesa
    for x in card
        p x
        if x.index(4)
            $cartinhas.append(x)
        end
    end
    p $cartinhas
end


mao1()
mao2()
dealer()
tonhos()


#puts"     ###  ###  ###
     #?#  #?#  #?#
     ###  ###  ###
     
        ###  ###
        #?#  #?#
        ###  ###"
pop()
pop()
pop()


#p maozinha2


#p maozinha1.index("♣️")



=begin




for cartinha in baralho
    if cartinha.include?("♣️") >= 5;
        p "aaa"
    end
end




=begin

for cartas in baralho
    print cartas
end

maozinha1 = baralho
    maozinha1.sort()
    maozinha1 = baralho.pop
    print maozinha1


def maozinha1(num)
    for x in range(num)
        carta_A = random.choice(baralho)
        maozinha1.append(carta_A)
        baralho.remove(carta_A)

carta 
=end
